buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy

```diff
@@ -146,31 +147,45 @@ class TestProgressLogger implements AggregatedEventListener {
             throw new IllegalArgumentException("Unknown test status: [${e.status}]")
         }
         testLogger.progress("Tests: completed: ${testsCompleted}, failed: ${testsFailed}, ignored: ${testsIgnored}")
-        String testName = simpleName(e.description.className) + '.' + e.description.methodName
+        String testName = testName(e.description)
         slaveLoggers[e.slave.id].progress("J${e.slave.id}: ${testName} ${statusMessage}")
     }
 
     @Subscribe
     void onTestStarted(TestStartedEvent e) throws IOException {
-        String testName = simpleName(e.description.className) + '.' + e.description.methodName
+        String testName = testName(e.description)
         slaveLoggers[e.slave.id].progress("J${e.slave.id}: ${testName} ...")
     }
 
     @Subscribe
     void onHeartbeat(HeartBeatEvent e) throws IOException {
-        String testName = simpleName(e.description.className) + '.' + e.description.methodName
+        String testName = testName(e.description)
         String time = formatDurationInSeconds(e.getNoEventDuration())
         slaveLoggers[e.slave.id].progress("J${e.slave.id}: ${testName} stalled for ${time}")
     }
 
+    /**
+     * Build the test name in the format of <className>.<methodName>
+     */
+    private static String testName(Description description) {
+        String className = simpleName(description)
+        if (description == null) {
+            return className
+        }
+        return className + "." + description.methodName
+    }
+
     /**
      * Extract a Class#getSimpleName style name from Class#getName style
      * string. We can't just use Class#getSimpleName because junit descriptions
      * don't always set the class field but they always set the className
      * field.
      */
-    private static String simpleName(String className) {
-        return className.substring(className.lastIndexOf('.') + 1)
+    private static String simpleName(Description description) {
+        if (description == null) {
+            return "<unknown>"
```

Comment:
Maybe this should indicate unknown class, and in the test name we should add unknown test?

Reply:
My thinking is if the description is null then we cannot know neither.
I believe, there is no case that the class is unknown but the method isn't or the opposite.

Reply:
I mean that we should be more explicit with unknown class vs unknown test.